### PR TITLE
GSoD2020 new Interactive Book structure

### PR DESCRIPTION
**Pull request by danielluehr**
This PR includes the proposed new structure of the book. 

All chapters (directories) and sections (md files in each chapter directory) contain only minimal content:

- [x] Front matter
- [x] Main heading
- [x] Chapter TOC (pointing to sections, generated automatically using jeckyll's liquid and JTD structure)
- [x] Section TOC code ("internal" jekyll/JTD processing)

Also, a new utility directory was created. It includes a python script which can generate the chapter and sections directory and files from a chapter template written in yaml (examples are included). It also include some helper emacs' lisp code to use org-mode source files to generate the MD files. Org sources can also be created using the python script.
- [x] chapter creation script
- [x] chapter template yaml file
- [x] example yaml files (used to populate the new proposed structure of the book)
- [x] org-mode helper lisp code

### CODE REVIEW

**Review by Shivansh2407 — CHANGES_REQUESTED** [review]
@danielluehr the changes look good to me. Nice work done 👍 . 

Some Things that need to be fixed : 

1. The Alignment of Headings in Sidebar with down arrow and without down arrow is different. 

2. The Edit on CircuitVerse currently is shown at the Bottom, shouldn't it be the same as shown in the top right as a button. Let me know your views.

3. The Conflicting files needs to fixed. I guess this has happened because Daniel has not pulled from the Master again after the first PR got merged in GSOD2020 branch.

4. The previous and the next buttons have been removed. Any specific reasons?

For all the above things mentioned, we can open up new issues and put them in the slack channel so that the new/existing contributors can help us with this. This way Daniel can focus more on the Documentation work and creating New content for the Book 📖 

@satu0king Please have a look and let me know your views

File: docs/binary-algebra/ieee-std-754.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+layout: circuitverse
+title: IEEE Std 754
+nav_order: l2s002
+cvib_level: advanced
+parent: Binary algebra
+has_children: false
+---
+
+# IEEE Standar 754
```
[the remark this answers is not in the document]
Comment (Shivansh2407) [reply docs/binary-algebra/ieee-std-754.md:10]: Same goes for utils/chapters/ch_binary_algebra.yml : Line 35

[the remark this answers is not in the document]
Comment (danielluehr) [reply docs/binary-algebra/ieee-std-754.md:10]: Hi, 

Sorry for creating the conflicts... I was so anxious to upload the changes that I forgot to merge upstream first...

I am submitting a new PR with the merge conflicts solved (I hope). Also, it includes the fix for the "Standard" typo.

Regarding the other comments I just want to provide complementary information:
1.  The current version (pre-GSoD2020) didn't use 2-level (i.e. chapters and sections), maybe it wasn't implemented in JTD at the time, but now it is a feature of JTD, but it needs proper styling in the navbar. Also the sections in the chapter (when the navbar is expanded) don't seem to have a consistent style (they are plain gray). It is the same in the current version (check the Flip-flops page and sub-pages, for an example). I will open an issue so anyone available and interested in fixing the styling can work on it.
2. I kept the "original" JTD "edit on github" link (which goes at the bottom) and added a link to the github repository on top. The idea is to stop breaking things with the upstream JTD Gem. In any case the links at the top need to get proper styling.
However, if you want to move the edit on github button to the top, I would recommend it to go in the same button bar as the sidebar button, and the code can be put in the corresponding include file.
3. (already mentioned above) 
4. The Prev/Next buttons liquid code started failing with the current JTD theme. I haven't investigated the reason. And it completely failed (jekyll wouldn't generate the site) when I added the full new structure (with chapters and sections). I will also open an issue so someone can start working on it and fix the prev/next navigation buttons.

Thanks!

**Review by Shivansh2407 — APPROVED** [review]
Working Well @danielluehr 🎉. I agree with you that we can open issues for the GSOD 2020 branch so that other contributors can also fix them while you can focus on content creation. 
#475 and #476 were opened earlier. 
Another possible issue is, when the sidebar button is clicked all the elements get shifted to left, while only the breadcrumb navigation element remains there. This should also sync with other elements when shifting to the left is what I think. let me know your views.
@satu0king please review this and merge accordingly 👍

**Comment by danielluehr** [comment]
> Working Well @danielluehr tada. I agree with you that we can open issues for the GSOD 2020 branch so that other contributors can also fix them while you can focus on content creation.
> #475 and #476 were opened earlier.
> Another possible issue is, when the sidebar button is clicked all the elements get shifted to left, while only the breadcrumb navigation element remains there. This should also sync with other elements when shifting to the left is what I think. let me know your views.
> @satu0king please review this and merge accordingly +1

About the sidebar alignment, you are right. I just copied the positioning calculations from the previous style to the new one and I think it needs double checking. Also, the breadcrumb was not used in the previous design, so its positioning was not updated, I think it is safe to style it accordingly and later it can be decided whether to keep or not the breadcrumbs.

Thanks!

**Pull request merged**